src/Compilers/Core/MSBuildTask/ManagedToolTask.cs

```diff
@@ -177,5 +175,46 @@ protected static ITaskItem[] GenerateCommandLineArgsTaskItems(List<string> comma
 
             return items;
         }
+
+        private string GetToolDirectory()
+        {
+            var buildTask = typeof(ManagedToolTask).Assembly;
+            var buildTaskDirectory = Path.GetDirectoryName(buildTask.Location)!;
```

Comment:
>! [](http://example.com/codeflow?start=78&length=1)

Should we guard against the null case? It seems like it could happen if we run this from the root directory or from memory (null location).